### PR TITLE
Add WWgg and ZZgg decays for HY production

### PR DESCRIPTION
**Pull request by chuwang1**
- Added H to WW, Y to GammaGamma decays for HY production.

- Added H to ZZ, Y to GammaGamma decays for HY production.

### CODE REVIEW

File: genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo2Q1L1Nu_YTo2G.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            '25:onMode = off', # Turn off all H decays
+            '25:oneChannel = 1 1 100 23 23', # H->ZZ
+            '25:onIfMatch = 23 23',
+            '35:onMode = off',
+            '35:oneChannel = 1 1 100 22 22',  # Y->gamma gamma
+            '35:onIfMatch = 22 22',
+            '15:onMode = on', # allow all tau decays. Leptonic and Hadronic
+            '23:mMin = 0.05', # the lower limit of the allowed mass range generated by the Breit-Wigner (in GeV)
+            '23:onMode = off', # Turn off all Z decays
+            '23:onIfAny = 1 2 3 4 5 11 13 15', # Add Z->enu, Z->munu, Z->taunu. Add Z->qq decays
```
[the remark this answers is not in the document]
Comment (chuwang1) [reply genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo2Q1L1Nu_YTo2G.py:27]: Thanks

File: genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo4Q_YTo2G.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            '35:onMode = off',
+            '35:oneChannel = 1 1 100 22 22',  # Y->gamma gamma 
+            '35:onIfMatch = 22 22',
+            '15:onMode = on', # allow all tau decays. Leptonic and Hadronic
+            '23:mMin = 0.05', # the lower limit of the allowed mass range generated by the Breit-Wigner (in GeV)
+            '23:onMode = off', # Turn off all Z decays
+            '23:onIfAny = 1 2 3 4 5', # Z->qq decays
+            'ResonanceDecayFilter:filter = on',
+            'ResonanceDecayFilter:exclusive = on', #off: require at least the specified number of daughters, on: require exactly the specified number of daughters
+            'ResonanceDecayFilter:udscAsEquivalent  = on',  #on: treat udsc quarks as equivalent
```
[the remark this answers is not in the document]
Comment (chuwang1) [reply genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo4Q_YTo2G.py:30]: Thanks

File: genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo2L2Nu_YTo2G.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            '25:onMode = off', # Turn off all H decays
+            '25:oneChannel = 1 1 100 23 23', # H->ZZ
+            '25:onIfMatch = 23 23',
+            '35:onMode = off',
+            '35:oneChannel = 1 1 100 22 22',  # Y->gamma gamma 
+            '35:onIfMatch = 22 22',
+            '15:onMode = on', # allow all tau decays. Leptonic and Hadronic
+            '23:mMin = 0.05', # the lower limit of the allowed mass range generated by the Breit-Wigner (in GeV)
+            '23:onMode = off', # Turn off all Z decays
+            '23:onIfAny = 11 13 15', # Add Z->enu, Z->munu, Z->taunu. Add Z->qq decays
```
[the remark this answers is not in the document]
Comment (chuwang1) [reply genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo2L2Nu_YTo2G.py:27]: Thanks

File: genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo4Q_YTo2G.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (chuwang1) [reply genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo4Q_YTo2G.py]: Change `ResonanceDecayFilter:udscAsEquivalent  = on` to `ResonanceDecayFilter:udscbAsEquivalent  = on`

File: genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo2Q1L1Nu_YTo2G.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (chuwang1) [reply genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo2Q1L1Nu_YTo2G.py:27]: Removed this file

File: genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo4Q_YTo2G.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (chuwang1) [reply genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo4Q_YTo2G.py]: For WWto4Q, we don't need to set` ResonanceDecayFilter:udscbAsEquivalent  = on`?

File: genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo2L2Nu_YTo2G.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (menglu21) [reply genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo2L2Nu_YTo2G.py:27]: plz modify the fragment name also

File: genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo2L2Nu_YTo2G.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            'ResonanceDecayFilter:filter = on',
+            'ResonanceDecayFilter:exclusive = on', #off: require at least the specified number of daughters, on: require exactly the specified number of daughters
+            'ResonanceDecayFilter:eMuTauAsEquivalent = on', #on: treat electrons, muons , and taus as equivalent
+            'ResonanceDecayFilter:allNuAsEquivalent  = on', #on: treat all three neutrino flavours as equivalent
+            'ResonanceDecayFilter:udscAsEquivalent  = on',  #on: treat udsc quarks as equivalent
+            'ResonanceDecayFilter:mothers = 35,25,23', #list of mothers not specified -> count all particles in hard process+resonance decays (better to avoid specifying mothers when including leptons from the lhe in counting, since intermediate resonances are not gauranteed to appear in general
+            'ResonanceDecayFilter:daughters = 22,22,11,12,11,12', # gamma gamma ,lnu,lnu
```
Comment (menglu21) [inline genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo2L2Nu_YTo2G.py:34]: if you only want 4L, the filter is not correct here

File: genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo2Q1L1Nu_YTo2G.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (menglu21) [reply genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo2Q1L1Nu_YTo2G.py:27]: it seems the file is not removed from your repo

File: genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo4L_YTo2G.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            '23:mMin = 0.05', # the lower limit of the allowed mass range generated by the Breit-Wigner (in GeV)
+            '23:onMode = off', # Turn off all Z decays
+            '23:onIfAny = 11 13 15', #  only leptonic Z decays
+            'ResonanceDecayFilter:filter = on',
+            'ResonanceDecayFilter:exclusive = on', #off: require at least the specified number of daughters, on: require exactly the specified number of daughters
+            'ResonanceDecayFilter:eMuTauAsEquivalent = on', #on: treat electrons, muons , and taus as equivalent
+            'ResonanceDecayFilter:allNuAsEquivalent  = on', #on: treat all three neutrino flavours as equivalent
+            'ResonanceDecayFilter:udscbAsEquivalent  = on',  #on: treat udscb quarks as equivalent
+            'ResonanceDecayFilter:mothers = 35,25,23', #list of mothers not specified -> count all particles in hard process+resonance decays (better to avoid specifying mothers when including leptons from the lhe in counting, since intermediate resonances are not gauranteed to appear in general
+            'ResonanceDecayFilter:daughters = 22,22,11,12,11,12', # gamma gamma ,lnu,lnu
```
Comment (menglu21) [inline genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo4L_YTo2G.py:34]: for 4L, it should be 22 22 11 11 11 11

Comment (chuwang1) [reply genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo4L_YTo2G.py]: Thanks，fixed

File: genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo2L2Nu_YTo2G.py (hunk 4 above)
Comment (chuwang1) [reply genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo2L2Nu_YTo2G.py]: Removed this file, updated it to `ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo4L_YTo2G.py`

File: genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo2L2Nu_YTo2G.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (chuwang1) [reply genfragments/ThirteenPointSixTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo2L2Nu_YTo2G.py]: Removed this file, updated it to ResonanceDecayFilter_example_HY_XToYH_HTo2ZTo4L_YTo2G.py

**Pull request merged**